modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/util/Utils.java

```diff
@@ -495,8 +501,10 @@ public static List<NameValuePair> getConsentRequiredClaimsFromResponse(HttpRespo
             consentRequiredClaims = joiner.toString();
         } else if (isNotBlank(mandatoryClaims)) {
             consentRequiredClaims = mandatoryClaims;
-        } else {
+        } else if (isNotBlank(requestedClaims)) {
             consentRequiredClaims = requestedClaims;
+        } else {
+             return extractConsentRequiredClaimsFromConsentPage(redirectUrl);
```

Comment:
Will this method handle the logic from 510 onwards?

Reply:
The logic inside the method  is different in this method than logic onwards 510. Move this below from https://github.com/wso2/product-is/pull/15583/commits/ad8981fd56065cf343bb2012db5dc602ae2bb019